webcurator-ui/src/views/target/target-tabs/TargetTabPanelGeneral/modals/TargetAddHarvestAuthModal.vue

```diff
@@ -1,88 +1,152 @@
 const loading = ref(false);
 
+const harvestAuths = ref<{ id: number, name: string, agent: '', permissionId: number, startDate: '', endDate: '', urlPatterns: [] }[]>([]);
+const filteredHarvestAuths = ref<{ id: number, name: string, agent: '', permissionId: number, startDate: '', endDate: '',  urlPatterns: []  }[]>([]);
+
 const searchTerm = ref('');
 
 const seed = ref();
 
-const fetch = () => {
-    const searchParams = {
-        offset: 0,
-        limit: 1024
+const prepareData = (data: HarvestAuth[]) => {
+  data.forEach((harvestAuth: HarvestAuth) => {
+    if (harvestAuth.authorisingAgents.length > 0) {
+      harvestAuth.authorisingAgents.forEach((authorisingAgent: any) => {
+        if (authorisingAgent.permissions.length > 0) {
+          harvestAuths.value.push({
+            id: harvestAuth.id,
+            name: harvestAuth.name,
+            agent: authorisingAgent.name,
+            permissionId: authorisingAgent.permissions[0].id,
+            startDate: authorisingAgent.permissions[0].startDate,
+            endDate: authorisingAgent.permissions[0].endDate,
+            urlPatterns: authorisingAgent.permissions[0].urlPatterns
+          });
+        }
+      })
     }
+  });
+  filteredHarvestAuths.value = harvestAuths.value;
+};
 
-    loading.value = true;
-
-    rest
-        .post('harvest-authorisations', searchParams, { header: 'X-HTTP-Method-Override', value: 'GET' })
-        .then((data: any) => {
-            harvestAuths.value = data['harvestAuthorisations']
-            filteredHarvestAuths.value = harvestAuths.value
-            loading.value = false
-        })
-        .catch((err: any) => {
-            console.log(err.message)
-            loading.value = false
-        })
-}
+const fetch = () => {
+  const searchParams = {
+    offset: 0,
+    limit: 1024,
+  };
+
+  loading.value = true;
+
+  rest
+    .post('harvest-authorisations', searchParams, { header: 'X-HTTP-Method-Override', value: 'GET' })
+    .then((data: any) => {
+      returnedHarvestAuths.value = data['harvestAuthorisations'];
+      prepareData(returnedHarvestAuths.value);
+      loading.value = false;
+    })
+    .catch((err: any) => {
+      console.log(err.message);
+      loading.value = false;
+    });
+};
 
 const search = () => {
-    const lowerCaseSearchTerm = searchTerm.value.toLowerCase();
-    filteredHarvestAuths.value = harvestAuths.value.filter((g: any ) => 
-        g.name.toLowerCase().includes(lowerCaseSearchTerm)
-    );
-}
+  const lowerCaseSearchTerm = searchTerm.value.toLowerCase();
+  filteredHarvestAuths.value = harvestAuths.value.filter((g: any) => 
+    g.name.toLowerCase().includes(lowerCaseSearchTerm) ||
+    g.urlPatterns.some((urlPattern: string) => {
+      // Ignore trailing slashes
+      const trimmedUrlPattern = urlPattern.replace(/\/$/, '');
+      const trimmedSearchTerm = lowerCaseSearchTerm.replace(/\/$/, '');
+      return trimmedUrlPattern.includes(trimmedSearchTerm);
```

Comment:
When search for Seed URL, the refered Harvest Authes were not filtered out. If take "Seed URL" as the search condition, will it filter out the URL Patterns which match the URL, or the items which had been refered by this Seed? From my understanding, it should be the first option.

Reply:
This requires a change to the API